src/core/functions/distribution/ProbabilityDensityFunction.h

```diff
@@ -119,7 +119,15 @@ void RevBayesCore::ProbabilityDensityFunction<valueType>::swapParameterInternal(
 template <class valueType>
 void RevBayesCore::ProbabilityDensityFunction<valueType>::update( void )
 {
-    dist->setValue( new valueType(x->getValue()) );
+     if constexpr (std::is_same_v<valueType, AbstractHomologousDiscreteCharacterData>) {
```

Comment:
One possibility is that this should be done for all things that inherit from Cloneable.   For example:

```
template <class mixtureType>
void RevBayesCore::MixtureDistribution<mixtureType>::redrawValue( void )
{
    if constexpr(std::is_base_of_v<Cloneable,mixtureType>)
    {
	delete this->value;
	this->value = simulate().clone();
    }
    else
	(*this->value) = simulate();
}
```

I think generally if the distribution is Cloneable, we use value->clone() to get a new value.  Is it OK to reuse the old value here?  I can see why we'd want to avoid duplicating a large object for no reason, but possible distributions assume that they own the value and can delete it.

Reply:
Extending to all Cloneable objects is an obviously good suggestion; I've done that.

Cloning also makes sense to me, and your rationale is consistent with the use of `new valueType`: presumably the probability distribution is going to try to balance this `new` with a `delete` somewhere?  I've added a `clone`, but am not confident that my implementation as `&x->clone()->getValue()` is correct here.

Reply:
Hmm... I think `this->value = &x->clone()->getValue()` is not right because we aren't holding a reference to the clone, but to its sub-object.  We will thus never free the clone, but we will try to free its sub-object.

I think that we would want `this->value = x->getValue().clone()` -- assuming that compiles.  In that case we directly track the cloned object and can `delete` it later.